infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftCheInstallationLocation.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift.environment;
+
+import javax.inject.Singleton;
+
+@Singleton
+public class OpenShiftCheInstallationLocation {
+  /** @return The name of the namespace where Che is installed */
+  public String getInstallationLocationNamespace() {
+    String kubernetesNamespace = System.getenv("KUBERNETES_NAMESPACE");
+    String podNamespace = System.getenv("POD_NAMESPACE");
```

Comment:
Probably we can move the logic of obtaining the env vars to `@PostConstruct` to avoid reading env vars on every call

Reply:
good idea

Reply:
can you inject them as `env. KUBERNETES_NAMESPACE` in the constructor?

Reply:
is this env vars are set by k8s or che-operator?

Reply:
the problem is that helm & operator are using different env vars (POD_NAMESPACE - helm / KUBERNETES_NAMESPACE - operator)

Reply:
Is it possible to change the helm chart to also call this `KUBERNETES_NAMESPACE` or does helm define this dynamically?

Reply:
I'm also voting for this to be somehow injected in a constructor or at least having a no-arg ctor determining the value from the env vars and another ctor initializing the namespace from a provided argument.

Also we should probably throw an exception if we're not able to determine the namespace. There's little point in continuing to deploy che if it won't be able to start workspaces..

Reply:
+ 1 for Helm update - https://github.com/eclipse/che/blob/master/deploy/kubernetes/helm/che/templates/deployment.yaml#L38
-1 for throwing the exception (there might be some cases when the namespace can not be obtained for any reason)

for this particular issue let's just do nothing if the namepsace can not be obtained - https://github.com/eclipse/che/pull/16532/files#r406635675

Reply:
Moved to `@PostConstruct`

Reply:
why not injection in the constructor?